### PR TITLE
chore: Update dependencies that `libraries.io` consider "out of date"

### LINKED ISSUE #6264: Update dependencies that libraries.io considers "out of date"
[According to Libraries.io we have 15 dependencies that are "out of date"](https://libraries.io/pypi/meltano)

These are:
- [x] `authlib`
- [x] `click`
- [x] `fasteners`
- [x] `flask`
- [x] `flask-executor`
- [x] `gunicorn`
- [x] `ipython`
- [x] `jsonschema`
- [x] `MarkupSafe`
- [x] `pypika`
- [x] `python-dotenv`
- [x] `python-gitlab`
- [x] `sqlparse`
- [x] `watchdog`
- [x] `werkzeug`

Where possible, we should update these dependencies. This may impact 
- https://github.com/meltano/meltano/issues/2830

### PR DESCRIPTION
**Pull request by WillDaSilva**
Relates to #2830, #3254, #5855

Closes #6264

### CODE REVIEW

**Review by edgarrmondragon — APPROVED** [review]
Ran some manual tests with the UI, `run` and `select` and things are looking good 😄

**Review by pandemicsyn — APPROVED** [review]
tl;dr Not apprehensive about potential impact to meltano core (e.g. click version bump) since we exercise things pretty well via tests. Little nervous about the API since it doesn't get very much attention by us.

I think I might have preferred this had been broken up across a couple PR's (meltano core impacting vs API impacting) so that a rollback in case of an issue was a bit easier and so that skimming the commit history was a bit more obvious. We should just keep an eye out for reports of flakey-ness in the UI/API in the near term.

**Comment by WillDaSilva** [comment]
> I think I might have preferred this had been broken up across a couple PR's (meltano core impacting vs API impacting) so that a rollback in case of an issue was a bit easier and so that skimming the commit history was a bit more obvious. We should just keep an eye out for reports of flakey-ness in the UI/API in the near term.

@pandemicsyn My apologies. I wasn't aware of the lack of test coverage there. Do we have tickets to track the improvement/addition of tests in this area?

**Comment by pandemicsyn** [comment]
> > I think I might have preferred this had been broken up across a couple PR's (meltano core impacting vs API impacting) so that a rollback in case of an issue was a bit easier and so that skimming the commit history was a bit more obvious. We should just keep an eye out for reports of flakey-ness in the UI/API in the near term.
> 
> @pandemicsyn My apologies. I wasn't aware of the lack of test coverage there. Do we have tickets to track the improvement/addition of tests in this area?

Yea, can't find it at the moment but sometime last year I snagged a coverage report that outlined where our gaps where. Theoretically, the API is also in maintenance-only mode pending us reworking the whole thing: https://github.com/meltano/meltano/blob/main/docs/adr/0005-use-fastapi-to-develop-v2-of-api.md#consequences

**Comment by pandemicsyn** [comment]
> Yea, can't find it at the moment but sometime last year I snagged a coverage report that outlined where our gaps where.

A found it - https://gitlab.com/meltano/meltano/-/issues/2959

**Comment by WillDaSilva** [comment]
Converted to draft until #6280 is addressed (in a different PR that I will open). Thanks @dingobar!

**Comment by edgarrmondragon** [comment]
:shipit:

**Comment by WillDaSilva** [comment]
> :shipit:

I had intended to wait for another approval, but you make a compelling argument :slightly_smiling_face:
